spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/SeparableConvolution2D.scala

```diff
@@ -0,0 +1,103 @@
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn.SpatialSeperableConvolution
+import com.intel.analytics.bigdl.nn.abstractnn._
+import com.intel.analytics.bigdl.optim.Regularizer
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, DataFormat}
+
+import scala.reflect.ClassTag
+
+class SeparableConvolution2D[T: ClassTag](val nbFilter: Int,
+                                          val nbCol: Int,
+                                          val nbRow: Int,
+                                          val depthMultiplier: Int = 1,
+                                          val subsample: (Int, Int) = (1, 1),
+                                          val borderMode: String = "valid",
+                                          val bias: Boolean = true,
+                                          val format: DataFormat = DataFormat.NCHW,
+                                          var depthwiseRegularizer: Regularizer[T] = null,
+                                          var bRegularizer: Regularizer[T] = null,
+                                          var pointwiseRegularizer: Regularizer[T] = null,
+                                          var inputShape: Shape = null
+  )(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Tensor[T], Tensor[T], T](KerasLayer.addBatch(inputShape)) {
+
+  require(borderMode.toLowerCase() == "valid" || borderMode.toLowerCase() == "same",
+    s"$borderMode is not supported")
+
+  override def doBuild(inputShape: Shape): AbstractModule[Tensor[T], Tensor[T], T] = {
+    val input = inputShape.toSingle().toArray
+
+    val stackSize = if (format == DataFormat.NCHW) input(1) else input(3)
+    val pad = KerasUtils.getPadsFromBorderMode(borderMode)
+
+    val layer = SpatialSeperableConvolution(
+      nInputChannel = stackSize,
+      nOutputChannel = nbFilter,
+      depthMultiplier = depthMultiplier,
+      kW = nbCol,
+      kH = nbRow,
+      sW = subsample._2,
+      sH = subsample._1,
+      pW = pad._1,
+      pH = pad._2,
+      hasBias = bias,
+      dataFormat = format,
+      wRegularizer = depthwiseRegularizer,
+      bRegularizer = bRegularizer,
+      pRegularizer = pointwiseRegularizer
+    )
+    layer.asInstanceOf[AbstractModule[Tensor[T], Tensor[T], T]]
+  }
+}
+
+
+object SeparableConvolution2D {
+  def apply[@specialized(Float, Double) T: ClassTag](
+    nbFilter: Int,
+    nbCol: Int,
+    nbRow: Int,
+    depthMultiplier: Int = 1,
+    subsample: (Int, Int) = (1, 1),
+    borderMode: String = "valid",
+    bias: Boolean = true,
+    format: DataFormat = DataFormat.NCHW,
+    depthwiseRegularizer: Regularizer[T] = null,
+    bRegularizer: Regularizer[T] = null,
+    pointwiseRegularizer: Regularizer[T] = null,
```

Comment:
use dRegularizer and pRegularizer, same as Keras

Reply:
Keras use depthwise_regularizer, pointwise_regularizer and b_regularizer either in documentation (https://faroit.github.io/keras-docs/1.2.2/layers/convolutional/#separableconvolution2d) and code (https://github.com/keras-team/keras/blob/1.2.2/keras/layers/convolutional.py)

Reply:
OK.